stormpiper/stormpiper/spa/src/components/tooltip.jsx

```diff
@@ -1,12 +1,30 @@
 // DeckGL react component
 const fieldDict = {
```

Comment:
this is inflexible and doesn't scale for new maps. refactor to pass fields as a prop rather than embedding a lookup table in the tooltip component. this component does not need to know the names of its users, just the tt label and the fields to report.

Reply:
I misleadingly gave this file a .jsx extension even though it doesn't use React at all. So this is actually not a component, just a helper function that performs the field lookup for the tooltip.

Given that the backend does not define the fields we want to display in the tooltip, and the getTooltip function is defined to receive a generic map object, we don't have the opportunity to explicitly pass fields as props/parameters. So I think we do need some lookup - it's just a matter of where we want to do it.

I think the most logical solution will be to put the helper function in map.jsx (it's not too big a file just yet), and get rid of tooltip.jsx.

Any thoughts?

Reply:
refactored